### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'lastname' when bringing up Inventory popup

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#2344]
_Imported from trac ticket 2344.
Created by **dwarner**
Opened in trac: May 21, 2013, 10:22:57 AM
Last modified in trac: May 21, 2013, 10:50:42 AM_
***

**Comment by kentbower** [comment]
_**kbower** trac comment on May 21, 2013, 10:39:11 AM:_

This is caused by corrupt data, in which the customer no longer exists for the order.

**Comment by kentbower** [comment]
_**kbower** trac comment on May 21, 2013, 10:47:46 AM:_

From Steinhafels train:

```
  1* select o.orderid, o.customerid 
from orders o, orderdetails l 
where o.orderid = l.orderid 
and l.productid = '404019019' 
and o.customerid is not null 
and not exists (select 0 from customers c 
where c.customerid = o.customerid)
SQL> /

ORDERID        CUSTOMERID
-------------- ----------
0805105LMIO    HAMRG38078
0420344YLYA    SEMIW23578

SQL> select * from customers where customerid in ('SEMIW23578','HAMRG38078')
  2  ;

no rows selected
```

**Comment by kentbower** [comment]
_**kbower** trac comment on May 21, 2013, 10:50:42 AM:_

(Changeset [[4156]](../commit/540461ffb1144aae5e45607af17ca744e6e2a9ce))
* closes #2344, allow Inventory popup to display even with corrupt data where an open order's customer no longer exists #stn

**Comment by kentbower** [comment]
_**kbower** trac status change on May 21, 2013, 10:50:42 AM:_ **closed**
